modules/bridge/src/main/ts/ephox/bridge/components/view/ViewButton.ts

```diff
@@ -0,0 +1,39 @@
+import { FieldSchema, StructureSchema } from '@ephox/boulder';
+import { Result } from '@ephox/katamari';
+
+import * as ComponentSchema from '../../core/ComponentSchema';
+
+export interface ViewNormalButtonSpec {
+  type: 'button';
+  text: string;
+  buttonType?: 'primary' | 'secondary';
+  onAction: () => void;
+}
+
+export type ViewButtonSpec = ViewNormalButtonSpec;
+
+export interface ViewNormalButton {
+  type: 'button';
+  text: string;
+  buttonType: 'primary' | 'secondary';
+  onAction: () => void;
+}
+
+export type ViewButton = ViewNormalButton;
+
+const normalButtonFields = [
+  FieldSchema.requiredStringEnum('type', [ 'button' ]),
+  ComponentSchema.text,
+  FieldSchema.defaultedStringEnum('buttonType', 'secondary', [ 'primary', 'secondary' ]),
```

Comment:
Should we put this (`buttonType`) in ComponentSchema now? Is it shared by dialog buttons?

Reply:
I moved this to ComponentSchema as a buttonType and shared them between dialog and this api.

Reply:
This doesn't look right? Previously one used to use a `Optional` and the other used a defaulted value. The former had this comment as well:
> // this should be defaulted to `secondary` but the implementation needs to manage the deprecation

Are we sure this is correct or didn't break anything? @TheSpyder might know there since it looks like he did that change.

Reply:
IIRC it was added as Optional since it needs to fallback to using `ComponentSchema.primary` if not set.

Reply:
Yeah maybe to risky I roll this back. Not really part of what I'm doing here anyway someone can log a tech dept.